src/deroff.rs

```diff
@@ -1028,17 +1044,83 @@ impl Deroffer {
         }
     }
 
-    fn do_line(&self, s: &str) -> bool {
-        match s.chars().nth(0) {
-            Some('.') | Some('\'') => !request_or_macro(s),
-            Some(c) => {
+    fn do_tbl(&mut self) -> bool {
+        match self.tblstate {
+            TblState::Options => {
+                while !self.s.is_empty() && !"\n;".contains(self.str_at(0)) {
+                    self.skip_leading_whitespace();
+
+                    if !self.str_at(0).chars().all(|c| c.is_alphabetic()) {
+                        self.skip_char(1);
+                    } else {
+                        // find first non-alphabetic character
+                        match self.s.chars().position(|c| !c.is_alphabetic()) {
+                            Some(idx) if self.s.get(idx..=idx) == Some("(") => {
+                                // self.s -> option '(' arg '(' rest
+                                let option = self.s.split_at(idx).0;
+                                let mut iter = self.s.split_at(idx + 1).1.splitn(2, '(');
+                                let arg = iter.next().unwrap_or_default();
+                                let rest = iter.next().unwrap_or_default();
+
+                                if option.to_lowercase() == "tab" {
+                                    self.tblTab = arg.get(0..1).unwrap_or_default().to_owned();
+                                }
+
+                                // Likely faster alternative to self.s = rest
+                                self.s.drain(self.s.len() - rest.len()..);
+                            }
+                            _ => self.s.clear(),
+                        }
+                    }
+                }
+                self.tblstate = TblState::Format;
+                self.condputs("\n");
+            }
+            TblState::Format => {
+                while !self.s.is_empty() && !".\n".contains(self.str_at(0)) {
+                    self.skip_leading_whitespace();
+                    if !self.str_at(0).is_empty() {
+                        self.skip_char(1);
+                    }
+                }
+
+                if self.str_at(0) == "." {
+                    self.tblstate = TblState::Data;
+                }
+                self.condputs("\n");
+            }
+            TblState::Data => {
+                if !self.tblTab.is_empty() {
+                    self.s = self.s.replace(&self.tblTab, "\n");
+                }
+
+                self.text();
+            }
+        }
+
+        true
+    }
+
+    fn do_line(&mut self) -> bool {
+        match self.s.bytes().next().unwrap_or_else(|| unreachable!()) {
```

Comment:
Why do we claim the `None` case is unreachable here? I usually only reserve `unreachable!` for cases where I can locally verify it or where there is something inherent in a type's guarantees that isn't already captured by the type system.

Just a panic (via `unwrap` or `expect`) makes more sense here. E.g.,
```suggestion
        match self.s.bytes().next().expect("`do_line` called when `self.s` was empty") {
```

Reply:
Because it really is unreachable.
https://github.com/rust-dc/fish-manpage-completions/pull/99/files#diff-5b0804df3203e0b4fc4f245618d61d53R1120

```rust
             self.s = line.to_owned() + "\n";
```
There will be at least `\n`.

Reply:
But yeah, that's what I mean. You have to rely on something that's not local to the function and not inherent to the type. If I as a dev at some point use this same function elsewhere in the module, there's nothing explicit to say that it's not meant to be called when `self.s` is non-empty.